### PR TITLE
upgrade to phoenix 1.4

### PR DESCRIPTION
**Pull request by fadeojo**
- upgrade to phoenix 1.4
- fixed issue where unauthenticated graph api route fails to render playground.

### CODE REVIEW

**Comment by fadeojo** [comment]
@danielberkompas I don't know why the build failed  as I am not able to view the semaphoreci page. Please advise.

File: template/$PROJECT_NAME$/apps/$PROJECT_NAME$_api/lib/$PROJECT_NAME$_api/context.ex [hunk 1]
```diff
@@ -38,7 +38,7 @@ defmodule <%= @project_name_camel_case %>API.Context do
       ["Bearer " <> token] ->
         %{token: %Token{token: token}}
       _ ->
-        %{}
+        %{token: "invalid"}
```
Comment (fadeojo) [inline template/$PROJECT_NAME$/apps/$PROJECT_NAME$_api/lib/$PROJECT_NAME$_api/context.ex]: `Accounts.authenticate(%{})` throws an exception. I changed `%{}` to `%{token: "invalid"}` to ensure an error returned by the `Accounts.authenticate`

**Comment by danielberkompas** [comment]
Hi @fadeojo, I need to move Mithril to TravisCI. I'll add it to my list. I don't have access to it either!

**Comment by fadeojo** [comment]
Hi @danielberkompas I will create a PR for the travis file. Please can you add this project to travis so once the `travis.yaml` is in, it will kick in.

**Comment by fadeojo** [comment]
@danielberkompas I fixed all the errors in the generated project and all tests are passing. I also included the travis config to this PR. Thanks.

**Comment by danielberkompas** [comment]
@fadeojo I added the project to travis, but you might need to push another commit to get Travis to run a build?

**Comment by fadeojo** [comment]
@danielberkompas 1 of the hound test cases is failing. it doesn't happen on my mac. I created a new semaphoreci [build](https://semaphoreci.com/fadeojo/mithril) for my fork and it works fine.

**Comment by danielberkompas** [comment]
@fadeojo What would I need to do to add semaphore from your account to this repo?

**Comment by fadeojo** [comment]
@danielberkompas 
- go to https://semaphoreci.com/ and login with your github account
- add this repository and semaphoreci
-  add a job to run the test command `bin/test`

**Pull request merged**